rpc/client/helpers.go

```diff
@@ -82,3 +84,83 @@ func WaitForOneEvent(c EventsClient, eventValue string, timeout time.Duration) (
 		return nil, errors.New("timed out waiting for event")
 	}
 }
+
+var (
+	// ErrClientRunning is returned by Start when the client is already running.
+	ErrClientRunning = errors.New("client already running")
+
+	// ErrClientNotRunning is returned by Stop when the client is not running.
+	ErrClientNotRunning = errors.New("client is not running")
+)
+
+// RunState is a helper that a client implementation can embed to implement
+// common plumbing for keeping track of run state and logging.
+type RunState struct {
```

Comment:
Should this be a part of the service package. It feels like the pattern of having generic service components close to implementations while also maintaining `service.Service`.

Reply:
Also: I would like to be clear that I think the implementation of this type (the one I wrote) is silly—we probably don't need any concurrency protection at all, but our tests don't exercise concurrency so I am playing it safe. I also agree with your comments elsewhere that this interface isn't ideal in several ways. RunState is a conservative compatibility shim to keep things upright until I can kick some more of the cockroaches out from under the rocks. 🙂